docs(ui): start ui 5 release post

ismay opened the pull request:
Closes #28

HendrikThePendric commented on _posts/2020-05-28-ui-5.md:
````diff
@@ -0,0 +1,298 @@
+    <MenuItem label="Item 2" />
+</Menu>
+
+// And this will render as a popup
+<FlyoutMenu>
+    <MenuItem label="Item 1" />
+    <MenuItem label="Item 2" />
+</FlyoutMenu>
+```
````
The reality is slightly more complicated than this. I think the text is fine, but the example could be tweaked a bit. Two points of importance:
- This comment not fully correct: `// And this will render as a popup`. In fact,  the "root" flyout-menu is not is a popoup, it is in a `Card`. Nested menus / submenus will fly-out in new Cards. 
- Since the current example is only one level deep, it doesn't really showcase the fly-out sub-menus.

Perhaps we could tweak like this:
```jsx
// This will render in a Card with width/height restrictions
<FlyoutMenu>
    <MenuItem label="Item 1">
        {/* This "sub-menu" will be rendered in a card in a popover */}
        <MenuItem label="Item 1a" />
        <MenuItem label="Item 1a" />
    </MenuItem>
    <MenuItem label="Item 2" />
</FlyoutMenu>
```

HendrikThePendric replied:
Oh, I hadn't noticed the section that follows and explains how to make sub-menus.... Upon reflection, perhaps just updating the comment above would be better...
So just change `// And this will render as a popup` to `// And this will render in a Card with width/height restrictions`

ismay replied:
What do you think of this: 512c686765c14890989f3f9a41192187b3c3441e

ismay commented on _posts/2020-05-28-ui-5.md:
```diff
@@ -0,0 +1,298 @@
+
+## Layering components
+
+We've changed the set of components used to produce various types of overlays. This is because the underlying logic has been improved and to clarify the scope of certain components that did a bit too much. We now have: `Layer`, `ComponentCover` and `CenterContent`. They replace the `Backdrop` and the `ScreenCover`, which have been removed.
```
Updated in bafc2f6eb3b22bf75299a111289aaaa77bea9cf8 & 44c0be50a419a2612bec0878df5bd3a8b30798cb

ismay commented on _posts/2020-05-28-ui-5.md:
```diff
@@ -0,0 +1,298 @@
+
+### Layer
+
+`Layer` is an overlay component that fills the entire viewport and allows you to stack various components on top of one another. The `Layer` accepts an `onClick` callback, so you can catch clicks on the background of whatever you're rendering. It also has a `translucent` prop, if you don't want the `Layer` to darken the background. An example:
```
Ah, I did not know it worked that way. That seems like it's easy to get wrong (translucent vs transparent). I'd have expected something like a `darken` prop for this behaviour. I'll update the text 👍

ismay replied:
Fixed in b41dfd6b9cb2035fafa5facfb9b23c296833cbef

ismay commented on _posts/2020-05-28-ui-5.md:
```diff
@@ -0,0 +1,298 @@
+
+### ComponentCover
+
+The `ComponentCover` is similar to the `Layer`, except `ComponentCover` only fills its parent, provided that the parent has a `relative` or `absolute` position. The `ComponentCover` also accepts an `onClick` and `translucent` prop, just like the `Layer`. An example of the `ComponentCover`:
```
Yeah I just wanted to be as explicit as possible, and thought that fixed and sticky didn't make much sense. But you're right, they're still technically allowed. I'll add them 👍

ismay replied:
Fixed in 855ba5ae1a7533e5716fc2768095e8167bc02d34

jenniferarnesen commented on _posts/2020-05-28-ui-5.md:
````diff
@@ -0,0 +1,300 @@
+</div>
+```
+
+### CenterContent
+
+CenterContent is a component that centers its children. It has a `position` prop which can be used to vertically align the children at the `top`, `middle` (default), or `bottom`. It can be useful when you want to render a loading spinner on top of your app for example:
````
Should be "CenteredContent"

ismay replied:
Thanks, fixed it.

Mohammer5 commented:
I think this is ok, maybe even a bit too in depth to my taste. The release docs that you've linked to already provide a very good overview, especially over breaking changes. It's good to mention the options/selected prop changes, but we could do that under 1 or 2 headlines and mention both  the transfer and select components there to reduce the blog posts content to a minimum.

What I'm missing is some kind of proposal how to integrate upgrading the library into the workflow.
So e. g. there should be a branch, branched off the master branch, where the library could be updated. Then all other branches wouldn't be blocked and "just" would have to deal with merge conflicts (which ideally aren't too many if code has been abstracted well) once the ui-update branch has been merged.

ismay commented:
> I think this is ok, maybe even a bit too in depth to my taste. The release docs that you've linked to already provide a very good overview, especially over breaking changes. It's good to mention the options/selected prop changes, but we could do that under 1 or 2 headlines and mention both the transfer and select components there to reduce the blog posts content to a minimum.

Thanks, I've updated the post. I've grouped the `selected` prop changes. I'm a bit hesitant to collapse it more. I'm trying to be very explicit, so that users of all skill levels can follow what we did. In my opinion our changelog is not very friendly (not even to me, as someone who worked on the changes), plus it's just plain wrong at times (with the changes that were later reverted). That's why I've put examples for most of the changes, etc. It might be a bit much for us, but I can imagine it'll be helpful for novices or those not as familiar with our libraries.

> What I'm missing is some kind of proposal how to integrate upgrading the library into the workflow.
> So e. g. there should be a branch, branched off the master branch, where the library could be updated. Then all other branches wouldn't be blocked and "just" would have to deal with merge conflicts (which ideally aren't too many if code has been abstracted well) once the ui-update branch has been merged.

I've added a note about this to the top.

I've made some other small changes as well. Let me know what you think!
